geonode/base/api/views.py

```diff
@@ -1525,3 +1528,71 @@ def base_linked_resources_payload(instance, user, params={}):
         ret.pop("WARNINGS")
 
     return ret
+
+
+class AssetViewSet(DynamicModelViewSet):
+    """
+    API endpoint that allows Assets to be viewed or edited.
+    """
+
+    authentication_classes = [SessionAuthentication, BasicAuthentication, OAuth2Authentication]
+    permission_classes = [IsAuthenticatedOrReadOnly, UserHasPerms]
+    filter_backends = [
+        DynamicFilterBackend,
+        DynamicSortingFilter,
+        DynamicSearchFilter,
+        # TODO: add filtering by owner / admin
+    ]
+    queryset = Asset.objects.all().order_by("-created")
+    serializer_class = AssetSerializer  # TODO: appropriate Serializer should be switched for each Asset instance
+    pagination_class = GeoNodeApiPagination
+
+    def list(self, request, *args, **kwargs):
+        """
+        Only for lists, allows access to Assets only to owned ones, or to all of them if the user is an admin
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+
+        user = request.user
+        is_admin = user.is_superuser if user and user.is_authenticated else False
+
+        if is_admin:
+            pass
+        elif user and user.is_authenticated:
+            queryset = queryset.filter(owner=user)
+        else:
+            queryset = queryset.none()
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def _get_file(self, request, pk, attachment: bool):
+        asset = get_object_or_404(Asset, pk=pk)
+        if bad_response := get_perms_response(request, asset):
```

Comment:
can't this be replaced by the `IsOwnerOrAdmin` permission class? If yes, let's improve the perms class (if needed)